From d5ea00ab62ec37ee2cede47a3bcb9fc3204c59de Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 18 Jul 2018 20:26:08 +0000
Subject: [PATCH] Fix the cpi uid parsing

Change backported to the v36 release from
https://github.com/lcp/efivar/commit/2050c87161de43545eae36d901623ab10ce83510.

This has caused clr-boot-manager failures due to not being able to
create an efivar entry for Clear Linux.
---
 src/linux-pci.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/linux-pci.c b/src/linux-pci.c
index 87878c3..6cb9310 100644
--- a/src/linux-pci.c
+++ b/src/linux-pci.c
@@ -104,9 +104,12 @@ parse_pci(struct device *dev, const char *current)
         rc = read_sysfs_file(&fbuf,
                              "devices/pci%04hx:%02hhx/firmware_node/uid",
                              root_domain, root_bus);
-        if ((rc <= 0 && errno != ENOENT) || fbuf == NULL)
+        if (rc <= 0 && errno != ENOENT)
                 return -1;
         if (rc > 0) {
+                if (fbuf == NULL) {
+                        return -1;
+                }
                 rc = sscanf((char *)fbuf, "%"PRIu64"\n", &acpi_uid_int);
                 if (rc == 1) {
                         dev->pci_root.pci_root_acpi_uid = acpi_uid_int;
-- 
2.18.0

